File: src/preludes/prelude.rb
require 'date'
require 'active_support/all'

# Time injection for testing: set KLANG_NOW environment variable
# Example: KLANG_NOW="2025-12-01T19:34:00" ruby -r ./prelude.rb test.rb
if ENV['KLANG_NOW'] && !ENV['KLANG_NOW'].empty?
  KLANG_FIXED_TIME = DateTime.parse(ENV['KLANG_NOW'])

  class DateTime
    class << self
      alias_method :_original_now, :now
      def now
        KLANG_FIXED_TIME
      end
    end
  end

  class Date
    class << self
      alias_method :_original_today, :today
      def today
        KLANG_FIXED_TIME.to_date
      end
    end
  end

  class Time
    class << self
      alias_method :_original_now, :now
      def now
        KLANG_FIXED_TIME.to_time
      end
    end
  end
end

File: src/preludes/prelude.js
// Dayjs with plugins for temporal operations
const _dayjs = require('dayjs');
const duration = require('dayjs/plugin/duration');
const isoWeek = require('dayjs/plugin/isoWeek');
const quarterOfYear = require('dayjs/plugin/quarterOfYear');
_dayjs.extend(duration);
_dayjs.extend(isoWeek);
_dayjs.extend(quarterOfYear);

// Time injection for testing: set KLANG_NOW environment variable
// Example: KLANG_NOW="2025-12-01T19:34:00" node test.js
let dayjs;
if (process.env.KLANG_NOW) {
  const fixedTime = _dayjs(process.env.KLANG_NOW);
  dayjs = function(input) {
    if (input === undefined) {
      return fixedTime.clone();
    }
    return _dayjs(input);
  };
  // Copy static methods from original dayjs
  Object.keys(_dayjs).forEach(key => {
    dayjs[key] = _dayjs[key];
  });
} else {
  dayjs = _dayjs;
}
